mypy/checkexpr.py

```diff
@@ -789,17 +789,22 @@ def check_typeddict_call_with_kwargs(
         context: Context,
         orig_callee: Type | None,
     ) -> Type:
-        if not (callee.required_keys <= set(kwargs.keys()) <= set(callee.items.keys())):
+        actual_keys = kwargs.keys()
+        found_set = set(actual_keys)
+        if not (callee.required_keys <= found_set <= set(callee.items.keys())):
```

Comment:
Do you really need to convert these to sets? IIRC dictionary `.keys()` already behave like sets.

Reply:
You're quite right. I had no idea.

Fixed in f7a7f4f84ac06e5dadc7d6ab05094cd2ca9cd782